utils/io/content/contentwriter.go

```diff
@@ -0,0 +1,135 @@
+package content
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"sync"
+
+	"github.com/jfrog/jfrog-client-go/utils"
+	"github.com/jfrog/jfrog-client-go/utils/errorutils"
+)
+
+const (
+	outputFilePattern      = "%s.*.json"
+	jsonArrayPrefixPattern = "  \"%s\": ["
+	jsonArraySuffix        = "]\n"
+)
+
+type ContentWriter struct {
+	arrayKey string
+	// The output data file path.
+	outputFile *os.File
+	// The chanel which from the output records will be pulled.
+	recordsChannel chan interface{}
+	isCompleteFile bool
+	errorsQueue    *utils.ErrorsQueue
+	runWaiter      sync.WaitGroup
+	once           sync.Once
+}
+
+func NewContentWriter(chanCapacity int, arrayKey string, isCompleteFile, useStdout bool) (*ContentWriter, error) {
+	var fd *os.File
+	var err error
+	if useStdout {
+		fd = os.Stdout
+	} else {
+		fd, err = ioutil.TempFile("", fmt.Sprintf(outputFilePattern, arrayKey))
+		if err != nil {
+			return nil, errorutils.CheckError(err)
+		}
+	}
+	self := ContentWriter{}
+	self.arrayKey = arrayKey
+	self.outputFile = fd
+	self.recordsChannel = make(chan interface{}, chanCapacity)
+	self.errorsQueue = utils.NewErrorsQueue(chanCapacity)
+	self.isCompleteFile = isCompleteFile
+	return &self, nil
+}
+
+func (rw *ContentWriter) SetArrayKey(arrKey string) *ContentWriter {
+	rw.arrayKey = arrKey
+	return rw
+}
+
+func (rw *ContentWriter) GetOutputFilePath() string {
```

Comment:
For consistency with the content reader, let's rename to GetFilePath(). The same goes for the outputFile field.

Reply:
outputFile -> *os.File
I can't change the name to `filePath` like the reader struct.

Reply:
Why @Or-Gabay?